src/arrayMethodJoin.js

```diff
@@ -5,9 +5,24 @@
  *
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function() {
-    // write code here
+  [].__proto__.join2 = function(arg) {
+    let resString = '';
+    if (arg === undefined) {
+      arg = ',';
+    }
+    if (arg === null) {
+      arg += '';
+    }
+    for (let item of this) {
+      if (this.indexOf(item) === this.length - 1 || !arg) {
```

Comment:
indexOf uses loop inside. So for every item you are running a loop. 

If there are N elements in the array, this is doing N^2 operations. 

Rewrite this loop to avoid doing N^2 operations.